scalar/timestamp.go

```diff
@@ -29,14 +29,26 @@ const (
 type Timestamp struct {
 	Valid bool
 	Value time.Time
+	Unit  arrow.TimeUnit
 }
 
 func (s *Timestamp) IsValid() bool {
 	return s.Valid
 }
 
-func (*Timestamp) DataType() arrow.DataType {
-	return arrow.FixedWidthTypes.Timestamp_us
+func (s *Timestamp) DataType() arrow.DataType {
+	switch s.Unit {
+	case arrow.Second:
+		return arrow.FixedWidthTypes.Timestamp_s
+	case arrow.Millisecond:
+		return arrow.FixedWidthTypes.Timestamp_ms
+	case arrow.Nanosecond:
+		return arrow.FixedWidthTypes.Timestamp_ns
+	case arrow.Microsecond:
+		return arrow.FixedWidthTypes.Timestamp_us
+	default:
+		panic("unknown timestamp unit")
+	}
```

Comment:
```suggestion
	return &arrow.TimestampType{Unit: s.Unit}
```

Reply:
This loses the timezone info and makes things more complicated (arrow.TypeEqual doesn't work)